src/Filesystem/Provider/DirectoryAccess.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+namespace ILIAS\Filesystem\Provider;
+
+use ILIAS\Filesystem\DTO\Metadata;
+use ILIAS\Filesystem\Exception\DirectoryNotFoundException;
+use ILIAS\Filesystem\Exception\FileAlreadyExistsException;
+use ILIAS\Filesystem\Exception\IOException;
+use ILIAS\Filesystem\Visibility;
+
+/**
+ * Interface DirectoryAccess
+ *
+ * Defines all directory access operations of the filesystem.
+ * Filesystem role interface.
+ *
+ * @author  Nicolas Schäfli <ns@studer-raimann.ch>
+ * @since 5.3
+ * @version 1.0
+ *
+ * @public
+ */
+interface DirectoryAccess {
+
+	/**
+	 * Create a new directory.
+	 *
+	 * Please note that the Visibility interface defines two constants PUBLIC_ACCESS and PRIVATE_ACCESS
+	 * to ease the development process.
+	 *
+	 * @param string $path          The directory path which should be created.
+	 * @param string $visibility    The visibility of the directory. Defaults to visibility public.
+	 *
+	 * @return void
+	 *
+	 * @throws IOException                  If the directory could not be created.
```

Comment:
Because it could have several reasons why the directory could not be created (security, filesystem related issues ...). Therefore, I decided to throw a more general IOException.